apps/web/lib/getMainDomainOrgRedirect.ts

```diff
@@ -0,0 +1,45 @@
+import type { IncomingMessage } from "http";
+
+import { WEBAPP_URL } from "@calcom/lib/constants";
+import logger from "@calcom/lib/logger";
+
+const log = logger.getSubLogger({ prefix: ["lib", "getMainDomainOrgRedirect"] });
+export const getMainDomainOrgRedirect = (
+  req: IncomingMessage | undefined,
+  subTeamSlugs: string[],
+  targetHostName: string
+) => {
+  if (!req) {
+    return null;
+  }
+
+  const protocol = WEBAPP_URL.startsWith("https://") ? "https" : "http";
+  const originalUrl = new URL(req.url!, `${protocol}://${req.headers.host}`);
```

Comment:
Will add a check for this (.url) too, thought since we are calling on page load -> getServerSideProps, if req is defined , then url should be present, checking for !req on first line